go/vt/vtgate/legacy_scatter_conn_test.go

```diff
@@ -0,0 +1,745 @@
+	require.Error(t, errors[0])
+	assert.Contains(t, errors[0].Error(), want)
+
+	// TransactionMode_SINGLE in txconn
+	sc.txConn.mode = vtgatepb.TransactionMode_SINGLE
+	session = NewSafeSession(&vtgatepb.Session{InTransaction: true})
+	_, errors = sc.ExecuteMultiShard(ctx, rss0, queries, session, false)
+	require.Empty(t, errors)
+	_, errors = sc.ExecuteMultiShard(ctx, rss1, queries, session, false)
+	require.Error(t, errors[0])
+	assert.Contains(t, errors[0].Error(), want)
+
+	// TransactionMode_MULTI in txconn. Should not fail.
+	sc.txConn.mode = vtgatepb.TransactionMode_MULTI
+	session = NewSafeSession(&vtgatepb.Session{InTransaction: true})
+	_, errors = sc.ExecuteMultiShard(ctx, rss0, queries, session, false)
+	require.Empty(t, errors)
+	_, errors = sc.ExecuteMultiShard(ctx, rss1, queries, session, false)
+	require.Empty(t, errors)
+}
+
+func TestAppendResult(t *testing.T) {
+	qr := new(sqltypes.Result)
+	innerqr1 := &sqltypes.Result{
+		Fields: []*querypb.Field{},
+		Rows:   [][]sqltypes.Value{},
+	}
+	innerqr2 := &sqltypes.Result{
+		Fields: []*querypb.Field{
+			{Name: "foo", Type: sqltypes.Int8},
+		},
+		RowsAffected: 1,
+		InsertID:     1,
+		Rows: [][]sqltypes.Value{
+			{sqltypes.NewVarBinary("abcd")},
+		},
+	}
+	// test one empty result
+	qr.AppendResult(innerqr1)
+	qr.AppendResult(innerqr2)
+	if len(qr.Fields) != 1 {
+		t.Errorf("want 1, got %v", len(qr.Fields))
+	}
+	if qr.RowsAffected != 1 {
+		t.Errorf("want 1, got %v", qr.RowsAffected)
+	}
+	if qr.InsertID != 1 {
+		t.Errorf("want 1, got %v", qr.InsertID)
+	}
+	if len(qr.Rows) != 1 {
+		t.Errorf("want 1, got %v", len(qr.Rows))
+	}
+	// test two valid results
+	qr = new(sqltypes.Result)
+	qr.AppendResult(innerqr2)
+	qr.AppendResult(innerqr2)
+	if len(qr.Fields) != 1 {
+		t.Errorf("want 1, got %v", len(qr.Fields))
+	}
+	if qr.RowsAffected != 2 {
+		t.Errorf("want 2, got %v", qr.RowsAffected)
+	}
+	if qr.InsertID != 1 {
+		t.Errorf("want 1, got %v", qr.InsertID)
+	}
+	if len(qr.Rows) != 2 {
+		t.Errorf("want 2, got %v", len(qr.Rows))
+	}
+}
+
+func newTestLegacyScatterConn(hc discovery.LegacyHealthCheck, serv srvtopo.Server, cell string) *ScatterConn {
+	// The topo.Server is used to start watching the cells described
+	// in '-cells_to_watch' command line parameter, which is
+	// empty by default. So it's unused in this test, set to nil.
+	gw := GatewayCreator()(ctx, hc, serv, cell, 3)
+	tc := NewTxConn(gw, vtgatepb.TransactionMode_TWOPC)
+	return NewLegacyScatterConn("", tc, gw, hc)
+}
+
+func newTestScatterConn(hc discovery.HealthCheck, serv srvtopo.Server, cell string) *ScatterConn {
+	// The topo.Server is used to start watching the cells described
+	// in '-cells_to_watch' command line parameter, which is
+	// empty by default. So it's unused in this test, set to nil.
+	gw := NewTabletGateway(ctx, hc, serv, cell)
+	tc := NewTxConn(gw, vtgatepb.TransactionMode_TWOPC)
+	return NewScatterConn("", tc, gw)
+}
+
```

Comment:
you moved it to separate file. Then this does not belong here.

Reply:
good point. we can fix it by moving it to the other test file.